test/utils.go

```diff
@@ -76,24 +72,54 @@ func createDaemonClientPair(t *testing.T) (*p2pd.Daemon, *p2pclient.Client, func
 	return daemon, client, closer
 }
 
-func createMockDaemonClientPair(t *testing.T) (*mockdaemon, *p2pclient.Client, func()) {
-	daemonPath, clientPath, dirCloser := createTempDir(t)
-	dmaddr, err := ma.NewComponent("unix", daemonPath)
-	if err != nil {
-		t.Fatal(err)
+type makeEndpoints func(t *testing.T) (daemon, client ma.Multiaddr, cleanup func())
+
+func makeTcpLocalhostEndpoints(t *testing.T) (daemon, client ma.Multiaddr, cleanup func()) {
+	daemon, err := ma.NewMultiaddr("/ip4/127.0.0.1/tcp/0")
+	require.NoError(t, err)
+	client, err = ma.NewMultiaddr("/ip4/127.0.0.1/tcp/0")
+	require.NoError(t, err)
+	cleanup = func() {}
+	return
+}
+
+func makeUnixEndpoints(t *testing.T) (daemon, client ma.Multiaddr, cleanup func()) {
+	daemonPath, clientPath, cleanup := createTempDir(t)
+	daemon, err := ma.NewComponent("unix", daemonPath)
+	require.NoError(t, err)
+	client, err = ma.NewComponent("unix", clientPath)
+	require.NoError(t, err)
+	return
+}
+
+var endpointRand = mathRand.New(mathRand.NewSource(1))
+
+// Pick a semi-random endpoint backend. May be timing dependent. Not good practice.
+func getEndpointsMaker(t *testing.T) makeEndpoints {
+	n := endpointRand.Intn(2)
```

Comment:
I don't understand why we do this. Can't we just deterministically test all endpoints that are valid for the OS?

Reply:
Why not just always use TCP on windows and unix sockets on unix? Wouldn't that end up testing the same thing without the nondeterminism?

Reply:
I think maybe that's the best solution.